core/pkg/server/unixlistener_common.go

```diff
@@ -0,0 +1,51 @@
+package server
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+)
+
+// listenInTempDir attempts to listen on a path constructed from os.TempDir().
+//
+// This is preferred over /tmp on macOS, but it may not work if the temporary
+// directory is very long because Unix sockets are generally limited to 92-108
+// characters.
+func listenInTempDir(namePattern string) (net.Listener, PortInfo, error) {
+	sockDir, err := makeUniqueDir(os.TempDir(), namePattern)
```

Comment:
should we allow users to configure the temp location?

Reply:
https://pkg.go.dev/os#TempDir

`$TMPDIR` (on Unix) or `%TMP%` (on Windows) is the way to go, if the user's system has restrictive permissions. I think otherwise, we could use the `root_dir` setting and write the port file there; we just don't pass it right now. The socket name length limit might be problematic in that case though.

Reply:
oh nice, yeah that makes sense...